src/gui/DatabaseWidget.cpp

```diff
@@ -1449,6 +1432,32 @@ void DatabaseWidget::search(const QString& searchtext)
     emit searchModeActivated();
 }
 
+void DatabaseWidget::saveSearch(const QString& searchtext)
+{
+    if (!m_db->isInitialized()) {
+        return;
+    }
+
+    // Pull the existing searches and prepend an empty string to allow
+    // the user to input a new search name without seeing the first one
+    QStringList searches(m_db->metadata()->savedSearches().keys());
+    searches.prepend("");
+
+    auto name = QInputDialog::getItem(
+        this, tr("Saved Search"), tr("Enter a unique name or overwrite an existing search from the list:"), searches);
```

Comment:
Can we make this an inline edit in the list of saved searches? We shouldn't be using 90's-style UI patterns.

Reply:
This is used on the first save of a search from the search bar. I am not going to change that pattern. You cannot rename a search.

Reply:
Sure, but I would add an inline edit field to the tag list whose contents are saved on enter. That would be a lot smoother. I just hate the modal popup.

Reply:
There are two problems solved with the modal popup:
1. When you press the save button on the far right, having a new line appear in the Searches and Tags view is undiscoverable and easily missed. The modal popup provides feedback and centrality to the feature.
2. The modal popup lets you choose an existing saved search from the list to overwrite. This is incredibly helpful if you want to tweak your search and keep the same name without typing it over again.

I would consider editing the name of a saved search from the Searches and Tags view, but that's about it. Same with tags, could bulk-edit the name of tag from that view as well which could be useful.

Reply:
Editing should definitely be done in-line. For creating, I'm on the fence. Every modal popup creates another point of conflict with database locking and its generally being quite obtrusive and old-school.

The locality issue is a valid concern, but that applies also to the added entry after you press Save. To solve these locality problems, you could maybe instead add a + button to the tag list, which shows a context menu that let's you create different kinds of tags, including saving the current search. It would be a little less discoverable, but it would solve those issues. An existing search could be updated by right-clicking and choosing "update search". In the grand scheme, I would actually prefer that, since it adds a central point where tags and searches are added and edited instead of having different actions all over the place. Then we could also implement things like adding tags to entries by drag and drop etc.

Reply:
I agree but that sounds like a great improvement for a future update. I am out of time for this one and it works great as is. Don't let perfect be the enemy of good enough.

Reply:
Nothing is more permanent than a temporary solution. 😉
Would the modal pop-up block the database lock timer?

Reply:
Yes it would and I can fix that easily, I am willing to do that work

Reply:
Done!